wagtail/admin/panels.py

```diff
@@ -770,13 +829,22 @@ def icon(self):
                 # "RegexField": "code",
                 # "BooleanField": "tick-inverse",
             }
-            field_type = self.bound_field.field.__class__.__name__
-
-            return self.panel.icon or field_icons.get(field_type, None)
+            return field_icons.get(form_field.__class__.__name__, None)
 
         def id_for_label(self):
+            # Although we want read-only output to be 'labelled', we aren't
+            # rendering an input, so do not want a <label> element.
+            if self.read_only:
+                return ""
             return self.bound_field.id_for_label
 
+        def heading_id(self):
+            # For read-only panels, we want to use the 'heading' as a label
+            # for the output (via 'aria-labelledby'), so provide this value
+            # for panel templates to pick up and apply.
+            if self.read_only:
+                return f"{self.prefix}-heading"
```

Comment:
Thanks @thibaudcolas, I'll explore this some more. From what I remember, using `id_for_label` resulted in some kind of unwanted output in one of the templates, but I might be able to get around that by changing what is included in the context.

Reply:
> From what I remember, using `id_for_label` resulted in some kind of unwanted output in one of the templates

It was `multi_field_panel_child.html` I was referring to. I ditched the returning so that read-only children had a `<h3>` instead of a `<label>` element. Have since updated the conditional to cater for the read-only case.

> I think this is off (unsure why). In my demo site, I get a different id and aria-labelledby

This should be sorted now for both 'top-level' panels and children of `MultiFieldPanel` :)